### PR TITLE
Weather System Implementation 

### PR DESCRIPTION
**Pull request by AzliRynz**
## Introduction
Adding `Weather System`

### API changes
Added `WeatherCommand`
Added permission `pocketmine.command.weather`
Added file in src/world/weather
* WeatherManager
* WeatherType

Added method in src/Server.php
* toggleWeather

Added method in src/world/World.php `getWeatherManager`
Requires translations:

## Tests

I tested this PR by doing the following (tick all that apply):
- [ ] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [ ] Playtesting using a Minecraft client (provide screenshots or a video)
- [ ] Writing a test plugin (provide the code and sample output)
- [ ] Other (provide details)

### CODE REVIEW

**Comment by GameParrot** [comment]
There should be an option in the config and a method to disable the weather cycle and an event for when weather changes
Lightning bolts should also be added

File: src/permission/DefaultPermissions.php [hunk 1]
```diff
@@ -106,6 +106,7 @@ public static function registerCorePermissions() : void{
 		self::registerPermission(new Permission(Names::COMMAND_UNBAN_IP, l10n::pocketmine_permission_command_unban_ip()), [$operatorRoot]);
 		self::registerPermission(new Permission(Names::COMMAND_UNBAN_PLAYER, l10n::pocketmine_permission_command_unban_player()), [$operatorRoot]);
 		self::registerPermission(new Permission(Names::COMMAND_VERSION, l10n::pocketmine_permission_command_version()), [$everyoneRoot]);
+		self::registerPermission(new Permission(Names::COMMAND_WEATHER, "Changing the weather in the world"));
```
Comment (Superomarking) [inline src/permission/DefaultPermissions.php]: ??

Comment (AzaleeX) [reply src/permission/DefaultPermissions.php]: a direct translation should be created `l10n::pocketmine_permission_command_weather()`

Comment (AzliRynz) [reply src/permission/DefaultPermissions.php]: I forgot to change the permission to the correct word

**Comment by ShockedPlot7560** [comment]
Why not using the world's gamerule?  It's not a plugin, you don't add as many config files as there are systems.

**Comment by AzliRynz** [comment]
Only temporary

**Pull request closed without merging**